website/content/docs/k8s/connect/ingress-controllers.mdx

```diff
@@ -0,0 +1,79 @@
+---
+layout: docs
+page_title: Ingress Controller Integrations
+description: Configuring Ingress Controllers With Consul On Kubernetes
+---
+
+# Configuring Ingress Controllers with Consul on Kubernetes
+
+-> This topic requires familiarity with [Ingress Controllers](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/)
+
+-> The following information requires Consul 1.10+, Consul-k8s 0.26+, Consul-helm 0.32+ configured with
+[Transparent Proxy](/docs/connect/transparent-proxy) mode enabled.
+
+This page describes a general approach for integrating Ingress Controllers with Consul on Kubernetes to secure traffic from the Controller
+to the backend services. This allows Consul to transparently secure traffic from the ingress point through the entire traffic flow of the service.
+
+A few steps are generally required to enable an Ingress controller to join the mesh and pass traffic through to a service:
+
+* Enable connect-injection via an annotation on the Ingress Controller's deployment: `consul.hashicorp.com/connect-inject` is `true`.
+
+* Using the following annotations on the Ingress controller's deployment, setup exclusion rules for it's ports.
+  * [`consul.hashicorp.com/transparent-proxy-exclude-inbound-ports`](https://www.consul.io/docs/k8s/connect#consul-hashicorp-com-transparent-proxy-exclude-inbound-ports) - Provides the ability to exclude a list of ports for
+inbound traffic that the service exposes from redirection. Typical configurations would require all inbound service ports
+for the controller to be included in this list.
+  * [`consul.hashicorp.com/transparent-proxy-exclude-outbound-ports`](https://www.consul.io/docs/k8s/connect#consul-hashicorp-com-transparent-proxy-exclude-outbound-ports) - Provides the ability to exclude a list of ports for
+outbound traffic that the service exposes from redirection.
+  * [`consul.hashicorp.com/transparent-proxy-exclude-outbound-cidrs`](https://www.consul.io/docs/k8s/connect#consul-hashicorp-com-transparent-proxy-exclude-outbound-cidrs) - Provides the ability to exclude a list of CIDRs that
+the service communicates with for outbound requests from redirection. It is somewhat common that an Ingress controller
+will expect to make API calls to the Kubernetes service for service/endpoint management. As such including the ClusterIP of the
+Kubernetes service is common.
```

Comment:
It'd be nice to mention for each of these when and why you'd need these. The example below doesn't exclude outbound ports, and it'd be nice to have some explanation for when you'd need to set it (if that's possible!).

Reply:
Good call! I don't think there are many cases for outbound port exclusions, at least none that I've found yet, aside from whats needed for kube api calls.
I can expand a bit for sure in the example. Just trying to keep it generic as possible.

Reply:
Makes sense! You have way more context on it, so if you think it won't makes sense because there are things that we don't know about ingress controllers, then feel free to ignore my suggestion! It was more of a nice to have 🙂